weblate/memory/models.py

```diff
@@ -280,6 +294,70 @@ def import_tmx(
                 found += 1
         return found
 
+    def import_other_format(
+        self,
+        request,
+        fileobj,
+        origin,
+        source_language: str | None = None,
+        target_language: str | None = None,
+        **kwargs,
+    ) -> int:
+        """
+        Import memory from other formats.
+
+        This is a generic function to import memories from other formats.
+        It currently supports all formats supported by `try_load` from
+        `weblate.formats.auto`.
+
+        """
+        from weblate.formats.auto import try_load
+
+        langcache = {}
+        try:
+            storage = try_load(origin, fileobj.read(), None, None)
+        except Exception as error:
+            report_error("Could not parse memory")
+            raise MemoryImportError(gettext("Unsupported file!")) from error
+
+        if storage.monolingual is True:
+            raise MemoryImportError(
+                gettext("Monolingual format not supported for memory upload")
+            )
+
+        def get_language(language: Language | str | None) -> Language:
```

Comment:
In method signature this is `str | None`, but here `Language | str | None`. Shouldn't these match?